server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -283,6 +285,16 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
 
         return properties;
     }
+
+    private errorSanitizer(error: any) {
```

Comment:
Instead of removing keys that we know might contain customer content, and running the risk of having other keys we were unaware of eventually leak more CC even after the fix, would it be appropriate to do the opposite: extract from the error only the keys we are interested in, which we know do not contain CC?

Reply:
Sounds like a nice idea, however in reality this will be hard to implement because of the deeply nested structure of Mongo error schema and large number of various fields. I think it would be an overkill. Removing just the target fields should be pretty safe, given that CC is potentially contained in those fields only and the Op schema does not change that often.

Reply:
I see. Let me try to put it in a different way. If I understand it correctly, besides telemetry, the error object is used by the mongoErrorRetryAnalyzer. Looking into a few of the rules, it seems that they usually depend only on a few of the error object's properties, which seem to be common across different types of mongo errors.

Example:
https://github.com/microsoft/FluidFramework/blob/94f6b19e1dcdc930f4ab0ea6052be6671191338f/server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoError/index.ts
- InternalErrorRule depends on error.code and error.codeName
- NoConnectionAvailableRule depends on error.message
- NoPrimaryInReplicasetRule depends on error.message
- PoolDestroyedRule depends on error.code and error.message
- RequestSizeLargeRule depends on error.code and error.errmsg
- RequestTimedNoRateLimitInfo depends on error.code and error.errmsg
- RequestTimedOutWithHttpInfo depends on error.code and error.errmsg
- and so on

Similarly, for: https://github.com/microsoft/FluidFramework/blob/94f6b19e1dcdc930f4ab0ea6052be6671191338f/server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoNetworkError/index.ts
- MongoNetworkTransientTransactionError depends on error.errorLabels

My suggestion was: given that we know exactly what properties we want to analyze, and that the properties above do not contain any customer content, then the errorSanitizer can explicitly extract only these properties from the original error. That will help both with telemetry, since the error code/error name/error message is already helpful for our purposes, as well as the retry logic, which will have all it needs to compute whether or not to retry. And at the same time, we would avoid running the risk of not omitting other fields that might eventually also contain CC.

Reply:
Sure. If the few error properties you mentioned are going to be enough for our purposes we can do that. I just thought there might be cases when having just those properties might not be enough. @zhangxin511 , @pradeepvairamani would appreciate your input on this as well.

Reply:
Got it. Yes, my assumption was that we only need those properties given the use cases of telemetry + deciding on whether we should retry. Feel free to dismiss my comment if that is not the case :)

Reply:
I agree that it'd be better to use a white list here to ensure we only take the properties that are safe (no CC) and are required for make other parts function properly. When we discover we need more data, we can always add to the white list. With a black list, we risk including the CC again, which is, as we know, very time-consuming to handle :smile:

Reply:
Compiled an allow list out of the keys required for error retry analysis and any other keys that do not contain CC. Updated logic error sanitization logic to be based on allow list.